tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+        parser.add_argument('--glc', 
+                    help='''
+                    Add the optional 3D glacier fields for verification of the glacier model.
+                    [default: %(default)s]
+                    ''', 
+                    action="store_true", 
+                    dest="glc_flag", 
+                    default=False)
+        parser.add_argument('--hirespft', 
+                    help='''
+                    If you want to use the high-resolution pft dataset rather
+                    than the default lower resolution dataset.
+                    (Low resolution is at quarter-degree, high resolution at 3-minute)
+                    [Note: hires only available for 1850 and 2005.]
+                    ''', 
+                    action="store_true", 
+                    dest="hres_flag", 
+                    default=False)
+        parser.add_argument('--crop', 
+                    help='''
+                    Create datasets with the extensive list of prognostic crop types.
+                    [default: %(default)s]
+                    ''', 
+                    action="store",
+                    type = str2bool,  
+                    nargs = '?',
+                    const = True,
+                    required = False,
+                    dest="crop_flag", 
+                    default=True)
+        parser.add_argument('-f','--fast', 
+                    help='Toggle fast mode which does not user the large mapping file', 
+                    action="store_true", 
+                    dest="fast_flag", 
+                    default=False)
+        parser.add_argument('-r','--res',
+                    help='''
+                    Resolution is the supported resolution(s) to use for files.
+                    [default: %(default)s]
+                    ''',
+                    action="store", 
+                    dest="res", 
+                    choices=valid_opts['res'], 
+                    required=False, 
+                    default="4x5")
+        return parser
+
+
+def str2bool(v):
+    """
+    Function for converting different forms of
+    command line boolean strings to boolean value.
+
+    Args:
+        v (str): String bool input
+
+    Raises:
+        if the argument is not an acceptable boolean string
+        (such as yes or no ; true or false ; y or n ; t or f ; 0 or 1).
+        argparse.ArgumentTypeError: The string should be one of the mentioned values.
+
+    Returns:
+        bool: Boolean value corresponding to the input.
+    """
+    if isinstance(v, bool):
+       return v
+    if v.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected. [true or false] or [y or n]')
+
+def tag_describe ():
+    """
+    Function for converting different forms of
+    command line boolean strings to boolean value.
```

Comment:
This is the description of a different function, so it needs to be replaced.

Reply:
Thanks. Fixed!